days/src/day11/Day11.kt

```diff
@@ -0,0 +1,95 @@
+package day11
+
+import Day
+import java.io.File
+
+class Day11(private val seatingMap: List<List<SeatStatus>>) : Day {
+    data class Coordinate(val row: Int, val column: Int)
+    enum class SeatStatus(val symbol: Char) { EMPTY('L'), FLOOR('.'), OCCUPIED('#'), NONE(' ') }
+
+    private fun List<List<SeatStatus>>.getStatus(seat: Coordinate) =
+            this.getOrElse(seat.row) {  listOf() }.getOrElse(seat.column) { SeatStatus.NONE }
+    private fun List<List<SeatStatus>>.isStatus(seat: Coordinate, status: SeatStatus) =
+            this.getStatus(seat) == status
+
+    private fun lineOfSight(seat: Coordinate, delta: Coordinate) =
+            generateSequence(Coordinate(seat.row+delta.row, seat.column+delta.column)) {
+                Coordinate(it.row + delta.row, it.column + delta.column)
+            }
+    private fun List<List<SeatStatus>>.inSight(seat: Coordinate, delta: Coordinate) =
+            lineOfSight(seat, delta).first { this.getStatus(it).let { status ->
+                status != SeatStatus.FLOOR
+            } }.let { this.getStatus(it) == SeatStatus.OCCUPIED }
+
+    private fun List<List<SeatStatus>>.adjacent(seat: Coordinate, delta: Coordinate) =
+            this.getStatus(lineOfSight(seat, delta).first()) == SeatStatus.OCCUPIED
+
+    private fun List<List<SeatStatus>>.occupiedSeats(seat: Coordinate, adjacentOnly: Boolean = true) =
+        (-1..1).fold(0) { sum, rowDelta ->
+            (-1..1).fold(sum) { rowSum, columnDelta ->
+                rowSum + if(adjacentOnly) {
+                    if (this.adjacent(seat, Coordinate(rowDelta, columnDelta))) { 1 } else { 0 }
```

Comment:
this if/else for return 1/0 comes up a few times, would be nice if there was a shorthand way to do it. Typically I would say return the boolean check coerced into an int but I'm not sure thats better